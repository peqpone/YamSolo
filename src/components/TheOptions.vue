<script setup lang="ts">
import useGameStore from '@/stores/game';
import { ref } from 'vue';
import RenderDie from '@/components/RenderDie.vue';

const themes = ['classic', 'design', 'numbers'];
const areOptionsVisible = ref<boolean>(false);
const gameStore = useGameStore();

const toggleOptions = () => {
  areOptionsVisible.value = !areOptionsVisible.value;
};
const setTheme = (newTheme:string) => {
  gameStore.saveTheme(newTheme);
};
</script>

<template>
  <div>
    <render-die
      class="die-preview"
      :die-value="1"
    />
    <a
      class="cta"
      @click="toggleOptions"
      @keydown="toggleOptions"
    >
      options
    </a>
    <div
      v-if="areOptionsVisible"
      class="options-container"
    >
      <render-die
        v-for="(theme, index) in themes"
        :class="`die-button theme-${theme}`"
        :key="index"
        @click="setTheme(theme)"
        :die-value="1"
        :force-theme="theme" />
    </div>
  </div>
</template>

<style lang="scss" scoped>
.die-preview {
  display: block;
  margin: 0 auto;
  width: 50px;
}
.die-button {
  cursor: pointer;
  width: 30px;
}
</style>
