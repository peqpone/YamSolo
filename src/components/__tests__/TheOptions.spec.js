import {
  describe, it, expect, beforeAll, vi, afterEach,
} from 'vitest';

import { shallowMount } from '@vue/test-utils';
import { createPinia, setActivePinia } from 'pinia';
import TheOptions from '../TheOptions.vue';
import useGameStore from '../../stores/game';

describe('TheOptions', () => {
  let gameStore;
  beforeAll(() => {
    setActivePinia(createPinia());
    gameStore = useGameStore();
    gameStore.reset();
    vi.spyOn(gameStore, 'saveTheme');
  });
  afterEach(() => {
    vi.clearAllMocks();
  });
  describe('Integration test', () => {
    let wrapper;
    it('should correctly mount initial state', () => {
      wrapper = shallowMount(TheOptions);
      expect(wrapper.html()).toMatchSnapshot();
    });
    it('should correctly render theme selection', async () => {
      await wrapper.find('.cta').trigger('click');
      expect(wrapper.html()).toMatchSnapshot();
    });
    it('choosing a theme should save it to the store', async () => {
      await wrapper.find('.theme-classic').trigger('click');
      expect(gameStore.saveTheme).toHaveBeenCalledWith('classic');
    });
  });
});
