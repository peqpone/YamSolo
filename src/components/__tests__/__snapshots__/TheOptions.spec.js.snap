// Vitest Snapshot v1

exports[`TheOptions > Integration test > should correctly mount initial state 1`] = `
"<div data-v-481f20da=\\"\\"><a class=\\"cta\\" data-v-481f20da=\\"\\"> options </a>
  <!--v-if-->
</div>"
`;

exports[`TheOptions > Integration test > should correctly render theme selection 1`] = `
"<div data-v-481f20da=\\"\\"><a class=\\"cta\\" data-v-481f20da=\\"\\"> options </a>
  <div class=\\"options-container\\" data-v-481f20da=\\"\\">
    <render-die-stub dievalue=\\"1\\" dieid=\\"0\\" forcetheme=\\"classic\\" class=\\"die-button\\" data-v-481f20da=\\"\\"></render-die-stub>
    <render-die-stub dievalue=\\"1\\" dieid=\\"1\\" forcetheme=\\"design\\" class=\\"die-button\\" data-v-481f20da=\\"\\"></render-die-stub>
    <render-die-stub dievalue=\\"1\\" dieid=\\"2\\" forcetheme=\\"numbers\\" class=\\"die-button\\" data-v-481f20da=\\"\\"></render-die-stub>
  </div>
</div>"
`;
